### DISCUSSION TITLE
'useFetch is undefined' error from ESLint

### DISCUSSION THREAD

**Discussion by chrissyast** [nuxt/nuxt#19572] [Questions]
### eslintrc.js

```
module.exports = {
  root: true,
  env: {
    browser: true,
    node: true
  },
  extends: [
    '@nuxtjs/eslint-config'
  ],
  plugins: [
  ],
  // add your custom rules here
  rules: {
    quotes: 'warn',
    indent: 'warn',
    'space-before-function-paren': 'off',
    'no-undef': 'warn'
  },
  parserOptions: {
    ecmaVersion: 'latest',
    sourceType: 'module'
  }
}

```

I can change the error to a warning to allow it to run, but why does eslint think that this is undefined?

**Comment by AloisSeckar** [comment]
For me usually helps to delete node_modules and package-lock.json and re-run `npm install` and `npm run dev` (or respective if using yarn or pnpm). Because those are often just some shenanigans from locally broken references.

If it doesn't help, we would probably need to see more of your code, to be able to spot the cause. Most probably it isn't problem in the linting itself.

**Reply by chrissyast** [reply]
Hi @AloisSeckar, thanks for your reply. Deleted and ran install again but still the same behaviour. https://github.com/chrissyast/eslint-problem

**Comment by Baroshem** [comment]
Hey, I believe that this error is actually coming from the problems with auto imports.

In Nuxt, you get the auto imported composables (such as useFetch) and you do not need to write the import statement. If you would like to use the unplugin (that does this magic) you need to also configure eslint to make it accessible:

```
  // Generate corresponding .eslintrc-auto-import.json file.
  // eslint globals Docs - https://eslint.org/docs/user-guide/configuring/language-options#specifying-globals
  eslintrc: {
    enabled: false, // Default `false`
    filepath: './.eslintrc-auto-import.json', // Default `./.eslintrc-auto-import.json`
    globalsPropValue: true, // Default `true`, (true | false | 'readonly' | 'readable' | 'writable' | 'writeable')
  },
```

I think that by having this local eslint files, you are overriding the default behavior.

Could you try to remove the eslint file and see if it makes a difference (to compare with default linter file from nuxt?)

**Reply by chrissyast** [reply]
Hi @Baroshem , into which file is the above code supposed to be pasted?

**Reply by Baroshem** [reply]
This code would be needed if you were using the unplugin auto import directly. But you are using nuxt which uses this plugin under the hood.

So you do not need to add it anywhere. I just think that the problem is with your eslint configuration

**Reply by chrissyast** [reply]
Ah ok I see. So I removed the eslint file and now the `useFetch is undefined` error disappears but now there is `Parsing error: Cannot use keyword 'await' outside an async function` on that same line. I had previously fixed that with the `parserOptions` property in the eslint file shown in my question (https://eslint.vuejs.org/user-guide/#using-eslint-v8-x)

**Reply by Baroshem** [reply]
Can you show your code where you are using the usefetch?

**Reply by chrissyast** [reply]
https://github.com/chrissyast/eslint-problem/blob/master/pages/index.vue

**Reply by chrissyast** [reply]
Hi @Baroshem . Do you have any idea what is causing this problem? It seems to be any of the helpers/methods provided by nuxt. It happens with `computed` and `navigateTo` too, and any composables I define.
